File: repos_estudiantes_TARQUI_primera_entrega/PRADO_FLORES_ARACELY_NAOMI/HERENCIA/HerenciaNueve/Politico.java
package HerenciaNueve;

public class Politico {
    protected String profesion;
    protected int añosExp;

    public Politico() {
        this.profesion = "Desconocida";
        this.añosExp = 0;
    }

    public Politico(String profesion, int añosExp) {
        this.profesion = profesion;
        this.añosExp = añosExp;
    }

    public void mostrar() {
        System.out.println("Profesión: " + profesion);
        System.out.println("Años de experiencia: " + añosExp);
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/HERENCIA/Ejercicio9/politico.py
class Politico:
    def __init__(self, profesion, anosExp):
        self.profesion = profesion
        self.anosExp = anosExp

class Partido:
    def __init__(self, nombreP, rol):
        self.nombreP = nombreP
        self.rol = rol

class Presidente(Politico, Partido):
    # SIN SOBRECARGA DE CONSTRUCTOR
    def __init__(self, nombre, apellido, profesion, anosExp, nombreP, rol):
        Politico.__init__(self, profesion, anosExp)
        Partido.__init__(self, nombreP, rol)
        self.nombre = nombre
        self.apellido = apellido

    def mostrar(self):
        print(f"Presidente: {self.nombre} {self.apellido}")
        print(f"Profesión: {self.profesion}, Años de experiencia: {self.anosExp}")
        print(f"Partido: {self.nombreP}, Rol en partido: {self.rol}")
def buscar_presidente_por_nombre(lista_presidentes, nombre_buscado):
    for p in lista_presidentes:
        if p.nombre.lower() == nombre_buscado.lower():
            print(f"\nPresidente encontrado: {p.nombre} {p.apellido}")
            print(f"Profesión: {p.profesion}")
            print(f"Partido Político: {p.nombreP}")
            return
    print("Presidente no encontrado.")


